Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -357,26 +272,153 @@ internal struct TextFormatScanner {
         return nil
     }
 
-    /// Assumes the leading quote has already been consumed
-    private mutating func parseQuotedString(terminator: UInt8) -> String? {
-        let start = p
-        while p != end {
-            let c = p[0]
-            if c == terminator {
-                let s = utf8ToString(bytes: start, count: p - start)
-                p += 1
-                skipWhitespace()
-                return s
+    /// Scan a string that encodes a byte field, return a count of
+    /// the number of bytes that should be decoded from it
+    private mutating func validateAndCountBytesFromString(terminator: UInt8) throws -> Int {
+      var count = 0
+      let start = p
+      while p != end {
+        let byte = p[0]
+        p += 1
+        if byte == terminator {
+          p = start
+          return count
+        }
+        switch byte {
+        case asciiBackslash: //  "\\"
+          if p != end {
+            let escaped = p[0]
+            p += 1
+            switch escaped {
+              case asciiZero...asciiSeven: // '0'...'7'
+                // C standard allows 1, 2, or 3 octal digits.
+                if p != end, p[0] >= asciiZero, p[0] <= asciiSeven {
+                  p += 1
+                  if p != end, p[0] >= asciiZero, p[0] <= asciiSeven {
+                    if escaped > asciiThree {
+                       // Out of range octal: three digits and first digit is greater than 3
+                      throw TextFormatDecodingError.malformedText
+                    }
+                    p += 1
+                  }
+                }
+                count += 1
+              case asciiLowerX: // 'x' hexadecimal escape
+                // C standard allows any number of digits after \x
+                // We ignore all but the last two
+                var hexDigits = 0
+                while p != end && fromHexDigit(p[0]) != nil {
```

Comment:
This seems wrong.  How would one use a hex digit and happen to need to follow it by something that is also hex?  "\x07for fun"

Tokenizer::ParseStringAppend() seems to say the C++ only accepts 1 or 2 following characters.